packages/flutter/lib/src/material/dialog.dart

```diff
@@ -44,6 +44,14 @@ const EdgeInsets _defaultInsetPadding = EdgeInsets.symmetric(horizontal: 40.0, v
 /// ** See code in examples/api/lib/material/dialog/dialog.0.dart **
 /// {@end-tool}
 ///
+/// {@tool snippet}
```

Comment:
Rather than a bare snippet, what do you think if we write it this way:
```suggestion
/// ## Contraints
/// The Material 3 guideline recommends that a dialog should have a maximal width of 560dp. 
/// For historical reasons, Flutter's [Dialog] widget does not come with this constraint by default.
/// For applications targeting large screens such as desktop or Web, it is recommended to 
/// set the [constraints] property.
/// {@tool snippet}
/// This sample shows a [Dialog] using [BoxConstraints] defined by the Material 3 specification.
///
/// ```dart
/// const Dialog(constraints: BoxConstraints(maxWidth: 560, minHeight: 280));
/// ```
/// {@end-tool}
```

Reply:
Also @QuncCccccc what do you think?

Reply:
I loved it. I'll commit it, and we can revert if Qunc has a different idea.